S/Singular/build_tarballs.jl

```diff
@@ -55,7 +80,7 @@ products = [
 # Dependencies that must be installed before this package can be built
 dependencies = [
     Dependency("cddlib_jll"),
-    Dependency("FLINT_jll"),
+    Dependency(PackageSpec(name="FLINT_jll", version=VersionSpec("200.690"))),
```

Comment:
Not totally sure what will happen to the compat entries but I think that at least the registry should be correct. I am still confused why and how exactly the two compat formats are different...

Reply:
Indeed, this is very confusing! But I compared the above with `semver_spec` and I *think* that means it is correct, i.e. will match exactly all version of the form `200.690.*`:
```julia
julia> Pkg.Types.semver_spec("200.690.0 - 200.690")
VersionSpec("200.690")

julia> v"200.690.0" in Pkg.Types.VersionSpec("200.690")
true

julia> v"200.690.1000" in Pkg.Types.VersionSpec("200.690")
true

julia> v"200.691.1000" in Pkg.Types.VersionSpec("200.690")
false
```

Of course whether it works right on the registry... we'll see!

@giordano could you please merge this (of course assuming it looks right to you)?

Reply:
Uhh, no, actually, please wait, I just thought of something...

Reply:
So the result is: 
- [registry](https://github.com/JuliaRegistries/General/blob/b169fc4a38b81d6e6eec7760600420bac2d28ffd/S/Singular_jll/Compat.toml#L6) `FLINT_jll = "200.690.0"`
- [project file](https://github.com/JuliaBinaryWrappers/Singular_jll.jl/blob/22ac1b95da33c159789235d5e27a2f33b1e833e1/Project.toml#L18) `FLINT_jll = "=200.690.0"`

Which is at least consistent. But both do not represent the given `VersionSpec` and is really not what I expected after https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/944, as `ranges[1].lower.n` should be 2 here.

Running what I think is the relevant function locally (using julia nightly and `.ci` as project) also gives something else:
```julia
julia> dependencies = [
           Dependency("cddlib_jll"),
           Dependency(PackageSpec(name="FLINT_jll", version=VersionSpec("200.690"))), # snapshot of 2.7.0-DEV
           Dependency("GMP_jll", v"6.1.2"),
           Dependency("MPFR_jll", v"4.0.2"),
       ];
julia> BinaryBuilder.build_project_dict("Singular",v"401.390.0",dependencies)
Dict{String, Any} with 5 entries:
  "deps"    => Dict{String, Any}("JLLWrappers"=>"692b3bcd-3c85-4b1f-b108-f13ce0eb3210", "Pkg"=>"44cfe95a-1eb2-52ea-b672-e2afdf69b78f", "cddlib_jll"=>"f07e07eb-5685-515a-97c8-3014f61…
  "name"    => "Singular_jll"
  "uuid"    => "43d676ae-4934-50ba-8046-7a96366d613b"
  "compat"  => Dict{String, Any}("JLLWrappers"=>"1.1.0", "julia"=>"1.0", "FLINT_jll"=>"200.690")
  "version" => "401.390.0"
```

Note that this is different but probably still wrong which I didn't realize when I tried to fix the `=` problem. It should be `~200.690` for FLINT (or maybe `200.690.0-200.690`, but that would require julia 1.4?).
The compat in the registry I think is the result from parsing the versions in Project.toml via `semver_spec` and then printing it again.

Reply:
*sigh* Thanks @benlorenz for checking this. Guess that means we need to tweak BB / BBB and possibly even more (registry code? pkg?). Annoying :-/. Maybe open a fresh issue for this (but where???)
CC @staticfloat @giordano